lib/rules/no-attrs-snapshot.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const utils = require('../utils/utils');
+
+//------------------------------------------------------------------------------
+// General rule -  Disallow use of attrs snapshot in `didReceiveAttrs`
+// and `didUpdateAttrs`.
+//------------------------------------------------------------------------------
+
+const message = 'Do not use the attrs snapshot that is passed in `didReceiveAttrs` and `didUpdateAttrs`. Please see the following guide for more information: https://github.com/ember-best-practices/eslint-plugin-ember-best-practices/blob/master/guides/rules/no-attrs-snapshot.md';
```

Comment:
Updated the URL. I like the idea of urls to rules docs. I think we should look at ways to assist end users as much as possible.